src/formatting/lists.rs

```diff
@@ -397,7 +405,9 @@ where
             item_max_width = None;
         }
 
-        if separate && sep_place.is_front() && !first {
+        if inner_item.is_empty() && item.pre_comment.is_some() && item.post_comment.is_some() {
```

Comment:
This is okay on its own, though along the same lines as the above comment I think declaring a local like `item_has_comments` and reusing could simplify some of these conditionals